maven-model-builder/src/main/java/org/apache/maven/model/building/FileSource.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.model.building;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class FileSource implements org.apache.maven.api.services.Source {
+    private final Path pom;
+
+    public FileSource(Path pom) {
+        this.pom = pom;
+    }
+
+    @Override
+    public Path getPath() {
+        return pom;
+    }
+
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return Files.newInputStream(pom);
+    }
+
+    @Override
+    public String getLocation() {
```

Comment:
Now I see the location is just the string form of the path. I'm not sure that's worth carrying around an extra method for. I suggest deleting getLocation.

Reply:
No, we may have other implementations such as:
 * https://github.com/apache/maven/blob/de19cfcd2bc8e774818d87472e8e64dc37c0b93d/maven-builder-support/src/main/java/org/apache/maven/building/StringSource.java
 * https://github.com/apache/maven/blob/de19cfcd2bc8e774818d87472e8e64dc37c0b93d/maven-builder-support/src/main/java/org/apache/maven/building/UrlSource.java